src/lib.rs

```diff
@@ -1410,6 +1419,14 @@ fn start_io(
             let inner_handle = expr_inner.0.start(context)?;
             return Ok(HandleInner::Unchecked(Box::new(inner_handle)));
         }
+        #[cfg(unix)]
```

Comment:
I'd lean towards leaving most of the new code in on all platforms, and only using conditional compilation in a couple places, one at the top in our public API, and one at the bottom where we call those functions that only exist on Unix.

Reply:
I made a commit that does this locally, however it causes this warning when compiling on windows:

```
warning: variant is never used: `Uid`
    --> src/lib.rs:1504:5
     |
1504 |     Uid(u32),
     |     ^^^^^^^^
     |
     = note: #[warn(dead_code)] on by default

warning: variant is never used: `Gid`
    --> src/lib.rs:1505:5
     |
1505 |     Gid(u32),
     |     ^^^^^^^^
```

Assuming you are okay with that, I will push my updated code.

Reply:
Oh good point. Hmm. We could suppress the lint explicitly on Windows. Or we could make the whole thing a runtime error... Now I don't feel strongly about any of the options :)

Reply:
I agree that sprinkling `#[cfg(unix)]` everywhere is annoying, but I think I prefer it to the other options.

I don't mind changing it though if you prefer.